mlflow/protos/webhooks.proto

```diff
@@ -0,0 +1,258 @@
+  flat_package: true,
+};
+
+// Webhook service for Model Registry
+service WebhookService {
+  rpc createWebhook (CreateWebhook) returns (CreateWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "POST",
+        path: "/mlflow/webhooks"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Create Webhook",
+    };
+  }
+
+  rpc listWebhooks (ListWebhooks) returns (ListWebhooks.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "GET",
+        path: "/mlflow/webhooks"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "List Webhooks",
+    };
+  }
+
+  rpc getWebhook (GetWebhook) returns (GetWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "GET",
+        path: "/mlflow/webhooks/{webhook_id}"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Get Webhook",
+    };
+  }
+
+  rpc updateWebhook (UpdateWebhook) returns (UpdateWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "PATCH",
+        path: "/mlflow/webhooks/{webhook_id}"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Update Webhook",
+    };
+  }
+
+  rpc deleteWebhook (DeleteWebhook) returns (DeleteWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "DELETE",
+        path: "/mlflow/webhooks/{webhook_id}"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Delete Webhook",
+    };
+  }
+
+  rpc testWebhook (TestWebhook) returns (TestWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "POST",
+        path: "/mlflow/webhooks/{webhook_id}/test"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Test Webhook",
+    };
+  }
+}
+
+// Webhook status enumeration
+enum WebhookStatus {
+  ACTIVE = 1;     // Webhook is active and receiving events
+  DISABLED = 2;   // Webhook is disabled and not receiving events
+  DELETED = 3;    // Webhook has been soft-deleted and is no longer active
+}
+
+// Webhook entity
+message Webhook {
+  // Unique identifier for the webhook
+  optional string webhook_id = 1;
+
+  // Name of the webhook
+  optional string name = 2;
+
+  // Optional description for the webhook
+  optional string description = 3;
+
+  // URL to send webhook events to
+  optional string url = 4;
+
+  // Optional secret key for HMAC signature verification
+  optional string secret = 5;
+
+  // List of event types this webhook is subscribed to
+  repeated string events = 6;
+
+  // Current status of the webhook
+  optional WebhookStatus status = 7;
+
+  // Timestamp when webhook was created
+  optional int64 creation_timestamp = 8;
+
+  // Timestamp when webhook was last updated
+  optional int64 last_updated_timestamp = 9;
```

Comment:
Custom header support is common (https://www.comet.com/docs/v2/guides/model-registry/webhooks/) in webhooks. We could add it if there is a feature request.

Reply:
I agree to have a custom header support when requested